### PR TITLE
feat: batches and start of worst case fees

### PR DESCRIPTION
**Pull request by QuantumExplorer**

### CODE REVIEW

File: grovedb/src/batch.rs [hunk 1]
```diff
@@ -530,10 +633,59 @@ impl GroveDb {
...
+        // nothing for the commit multi batch?
+        Ok(()).wrap_with_cost(cost)
+    }
 }
 
 #[cfg(test)]
 mod tests {
+    use std::option::Option::None;
+
+    use visualize::visualize_stderr;
```
Comment (fominok) [inline grovedb/src/batch.rs:687]: These two imports are not needed

File: grovedb/src/lib.rs [hunk 2]
```diff
@@ -64,6 +64,12 @@ pub enum Error {
     #[error("data corruption error: {0}")]
     CorruptedData(String),
 
+    #[error("corrupted code execution error: {0}")]
+    CorruptedCodeExecution(&'static str),
```
Comment (fominok) [inline grovedb/src/lib.rs:68]: What is that error about? Cannot see any usages

File: storage/src/rocksdb_storage/storage.rs [hunk 3]
```diff
@@ -82,6 +82,34 @@ impl RocksDbStorage {
         res = blake3::hash(&res).as_bytes().to_vec();
         res
     }
+
+    // /// A helper method to build a prefix to rocksdb keys or identify a subtree
```
Comment (fominok) [inline storage/src/rocksdb_storage/storage.rs:86]: I think this could be removed

**Review by fominok — APPROVED** [review]

File: storage/src/rocksdb_storage/storage.rs (hunk 3 above)
Comment (QuantumExplorer) [reply storage/src/rocksdb_storage/storage.rs:86]: Hmmmm, best to keep it for now, just in case.

**Pull request merged**